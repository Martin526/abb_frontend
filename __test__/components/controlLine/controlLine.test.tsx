import renderer, { ReactTestRenderer } from 'react-test-renderer';
import ControlLine from '../../../src/components/controlLine';
import React from 'react';
// import {cleanup, fireEvent, render} from '@testing-library/react';

describe('ControlLine test', () => {

	let controlLineRender: ReactTestRenderer;

	beforeEach(() => {
		controlLineRender = renderer.create(<ControlLine name='testname' dev={0} devOutTol={0} status='OK'/>);
	});

	afterEach(() => {
		controlLineRender.unmount();
	});

	test('check control line is render ok', () => {
		const tree = controlLineRender.toJSON();

		expect(tree).not.toBe(null);
		expect(tree).toMatchSnapshot();
	});

	test('check control line properties', () => {
		const root = controlLineRender.root;

		expect(root.props.name).toBe('testname');
		expect(root.props.dev).toBe(0);
		expect(root.props.devOutTol).toBe(0);
		expect(root.props.status).toBe('OK');
	});
});
