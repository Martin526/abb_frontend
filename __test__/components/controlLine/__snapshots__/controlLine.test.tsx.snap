// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlLine test check control line is render ok 1`] = `
<div
  className="sc-bdvvaa hIUbgN"
>
  <p
    className="sc-gsDJrp esAOwv"
  >
    testname
  </p>
  <p
    className="sc-gsDJrp esAOwv"
  >
    0
  </p>
  <p
    className="sc-gsDJrp esAOwv"
  >
    0
  </p>
  <p
    className="sc-gsDJrp esAOwv"
  >
    <svg
      className="sc-dkPtyc XyyyE"
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </p>
</div>
`;
